### PR TITLE
Add optional callback to forceUpdate method

### PR DESCRIPTION
**Pull request by wyze**
This PR aims to solve #434. I've also added a general `forceUpdate` is called and rerenders the component test since there didn't seem to be one. Let me know if I placed that in the right test file as I was unsure.

### CODE REVIEW

**Comment by developit** [comment]
Awesome, and it's only 7 bytes!

**Comment by NekR** [comment]
Callback is good, but I got to point out that if you are going to make forceUpdate() async someday, then that's major change (e.g. I rely on it to be sync).

**Comment by developit** [comment]
Agreed - I was going to suggest that might be worth bringing up, but I'm definitely relying on it being synchronous as well.  I don't think we can reasonably do that in the near term.

I'd be curious what the filesize would be with a more naive callback solution since it's always synchronous:

```js
forceUpdate(callback) {
  renderComponent(this, FORCE_RENDER);
  if (callback) callback();
}
```

(not suggesting that's the right way to go here, just crossed my mind)

**Comment by NekR** [comment]
@developit yeah, I thought about it too. There is no right way here, both are okay :-) (`callback && callback()` is smaller than `if` 😝)

File: test/browser/spec.js [hunk 1]
```diff
@@ -124,4 +124,49 @@ describe('Component spec', () => {
...
+			}
+			sinon.spy(ForceUpdateComponent.prototype, 'componentWillUpdate');
+			sinon.spy(ForceUpdateComponent.prototype, 'forceUpdate');
+			render(<ForceUpdateComponent />, scratch);
+			expect(ForceUpdateComponent.prototype.componentWillUpdate).not.to.have.been.called;
+
+			forceUpdate();
+
+			expect(ForceUpdateComponent.prototype.componentWillUpdate).to.have.been.called;
+			expect(ForceUpdateComponent.prototype.forceUpdate).to.have.been.called;
```
Comment (robertknight) [inline test/browser/spec.js]: Thanks for adding these tests, although it doesn't actually check that the component got re-rendered. Ideally the test would do something mirroring a simplified real-world use of `forceUpdate`. Something like:

1. Create a component whose output depends on something other than props/state
2. Render
3. Change the external state
4. Force update
5. Check that the rendered output reflects the updated external state.

File: test/browser/spec.js [hunk 2]
```diff
@@ -124,4 +124,49 @@ describe('Component spec', () => {
...
+		});
+
+		it('should add callback to renderCallbacks', () => {
```
Comment (robertknight) [inline test/browser/spec.js]: `renderCallbacks` is an implementation detail which we should avoid in the description. What the test actually _does_, checking that `forceUpdate` invokes its callback, is fine though.

**Pull request merged**